src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -182,46 +180,33 @@ private static InputTypeSerialization UpdateInputSerialization(InputModelType in
             return serialization;
         }
 
-        private static IReadOnlyList<InputModelProperty> UpdateInputModelProperties(InputModelType inputModel, INamedTypeSymbol? existingBaseType, string ns, SourceInputModel? sourceInputModel)
+        private IReadOnlyList<InputModelProperty> UpdateInputModelProperties(InputModelType inputModel, INamedTypeSymbol? existingBaseType, string ns, SourceInputModel? sourceInputModel)
```

Comment:
do we have to remove the static?
If we remove the static, quite a few parameters could be removed

Reply:
Remove static to be able to use instance fields, such as `typeFactory`, it does not need to be static

Reply:
See your point now, will update it